src/interpreter/executors/instruction.rs

```diff
@@ -499,6 +499,57 @@ where
     }
 }
 
+/// Computes nth root of target, rounding down to nearest integer.
+/// This function uses the floating point operation to get an approximate solution,
+/// but corrects the result using exponentation to check for inaccuracy.
+fn checked_nth_root(target: Word, nth_root: Word) -> Option<u64> {
+    if nth_root == 0 {
+        // Zeroth root is not defined
+        return None;
+    }
+
+    if nth_root == 1 || target <= 1 {
+        // Corner cases
+        return Some(target);
+    }
+
+    if nth_root >= target || nth_root > 64 {
+        // For any root >= target, result always 1
+        // For any n>1, n**64 can never fit into u64
+        return Some(1);
+    }
+
+    let nth_root = nth_root as u32; // Never loses bits, checked above
+
+    // Use floating point operation to get an approximation for the starting point.
+    // This is at most off by one in either direction.
+    let guess = (target as f64).powf((nth_root as f64).recip()) as u64;
+
+    debug_assert!(guess != 0, "This should never occur for {{target, n}} > 1");
+
+    // Compute guess**n to check if the guess is too large.
+    // Note that if guess == 1, then g1 == 1 as well, meaning that we will not return here.
+    let Some(g1) = guess.checked_pow(nth_root) else {
```

Comment:
I don't think we should jump to rust 1.65 features immediately, as there are several downstream deps that may not be ready for this change which will prevent a release.